src/zope/meta/setup_to_pyproject.py

```diff
@@ -0,0 +1,448 @@
+
+    if not path.is_dir():
+        print(f' - {path} is not a folder')
+        sane = False
+
+    if not (path / 'setup.py').exists():
+        print(' - no setup.py found, cannot convert package.')
+        sane = False
+
+    if not (path / '.meta.toml').exists():
+        print(' - no pyproject.toml found, cannot convert package.')
+        sane = False
+
+    return sane
+
+
+def main():
+    parser = get_shared_parser(
+        "Move package metadata from setup.py to pyproject.toml.",
+        interactive=True)
+    parser.add_argument(
+        '--dry-run',
+        dest='dry_run',
+        action='store_true',
+        default=False,
+        help='Do not make any changes but output contents for the changed'
+             ' setup.py and pyproject.toml files.',
+    )
+    args = parser.parse_args()
+
+    print(f'Converting package {args.path.name}')
+
+    if not package_sanity_check(args.path):
+        print('Conversion not possible, exiting.')
+        sys.exit()
+
+    (leftover_setup_kwargs, toml_dict) = parse_setup_py(args.path / 'setup.py')
+
+    # Sanity check - if project has been converted already, give up.
+    if 'name' not in toml_dict['project'] and \
+       'version' not in toml_dict['project']:
+        print('Package has been converted already, exiting.')
+        sys.exit()
+
+    toml_content = rewrite_pyproject_toml(args.path / 'pyproject.toml',
+                                          toml_dict)
+    setup_content = rewrite_setup_py(args.path / 'setup.py',
+                                     leftover_setup_kwargs)
+
+    # If this is a dry run, just print the end result and exit.
+    if args.dry_run:
+        print('\n------------> pyproject.toml with all changes applied:')
+        print(toml_content)
+        print('\n------------> setup.py with all changes applied:')
+        print(setup_content)
+        sys.exit()
+
+    with open(args.path / 'pyproject.toml', 'w') as fp:
+        fp.write(toml_content)
+    with open(args.path / 'setup.py', 'w') as fp:
+        fp.write(setup_content)
+
+    if args.interactive or args.commit:
+        print('Look through setup.py to see if it needs changes.')
+        call(os.environ['EDITOR'], 'setup.py')
+        print('Look through pyproject.toml to see if it needs changes.')
+        call(os.environ['EDITOR'], 'pyproject.toml')
+
+    with change_dir(args.path) as cwd:
+        if args.run_tests:
+            tox_path = shutil.which('tox') or (
+                pathlib.Path(cwd) / 'bin' / 'tox')
+            call(tox_path, '-p', 'auto')
+
+        branch_name = args.branch_name or "convert-setup-py-to-pyproject-toml"
+        updating = git_branch(branch_name)
+
+        if args.commit:
+            if args.commit_msg:
+                commit_msg = args.commit_msg
+            else:
+                commit_msg = ('Move package metadata from setup.py'
+                              ' to pyproject.toml.')
+            call('git', 'commit', '-m', commit_msg)
```

Comment:
Might we also add a change log entry like in https://github.com/zopefoundation/meta/blob/24cb1653095b1806e55bb9d9968ce848de801a10/src/zope/meta/pep_420.py#L47-L51?